doc/overlays.xml

```diff
@@ -96,4 +96,68 @@ overridden and/or new packages.</para>
 
 </section>
 
+<section xml:id="sec-overlays-extend-modify">
+<title>Overlays used to extend or modify pkgs</title>
+
+<para>
+If you want to extend nixpkgs with your own packages: add new packages, alter existing ones then you can use this code <filename>nixcloud-pkgs.nix</filename>:
+</para>
+
+<programlisting>
+{  pkgs ? import &lt;nixpkgs> {}
+,  nixcloud-backend   ? pkgs.stdenv.lib.cleanSource ../nixcloud-backend
+,  nixcloud-frontend  ? pkgs.stdenv.lib.cleanSource ../nixcloud-frontend
+,  nixcloud-editor    ? pkgs.stdenv.lib.cleanSource ../nixcloud-editor
+, ...
+}:
+let
+  nc-backend = nixcloud-backend;
+  nc-frontend = nixcloud-frontend;
+  
+  newpkgs = import pkgs.path { overlays = [ (pkgsself: pkgssuper: {
+    nixcloud-backend  = pkgs.callPackage nc-backend { inherit newpkgs; };
+    nixcloud-frontend = pkgs.callPackage "${nc-frontend}/release.nix" {
+      inherit nixcloud-editor;
+    };
+  } ) ]; };
+in newpkgs
+</programlisting>
+
+<para>
+If you want to install a package from this new set, do this:
+
+<programlisting>
+nix-build nixcloud-pkgs.nix -A nixcloud-backend
```

Comment:
Once you created the overlays, as I mentioned in the previous comment and installed it, then you should be able to install the package like this:

```shell
$ nix-build -A nixcloud-backend
```

Note, that overlays are extending Nixpkgs by the fact that Nixpkgs import the overlays, not the opposite.  Nixpkgs is made visible in your overlay code, by the mean of `self` and `super`, and thus should not be imported with `import <nixpkgs> {}` expression, which can be incorrect, as someone might be using `nixpkgs` in a cross compilation environment for example.